clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -42,8 +42,11 @@ public KafkaChannel(String id, TransportLayer transportLayer, Authenticator auth
     }
 
     public void close() throws IOException {
-        transportLayer.close();
-        authenticator.close();
+        try {
+            transportLayer.close();
+        } finally {
+            authenticator.close();
```

Comment:
This can also throw an exception. We should probably catch it and add the original exception as supposed (see `Throwable.addSuppressed`.